verifier/cosignerverifier.go

```diff
@@ -0,0 +1,94 @@
+package verifier
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jws"
+	"github.com/openpubkey/openpubkey/pktoken"
+	oidcclient "github.com/zitadel/oidc/v2/pkg/client"
+)
+
+type CosignerVerifier struct {
+	issuer  string
+	options CosignerVerifierOpts
+}
+
+type CosignerVerifierOpts struct {
+	// Strict specifies whether or not a pk token MUST contain a signature by this cosigner.
+	// Defaults to false.
+	Strict bool
+	// Allows users to set custom function for discovering public key of Cosigner
+	DiscoverPublicKey func(ctx context.Context, kid string, issuer string) (jwk.Key, error)
```

Comment:
This pattern here is really nice. I can't wait to use it to plug my historical public key package into cosigner.